addons/necropost-highlighter/userscript.js

```diff
@@ -0,0 +1,307 @@
+/**
+ * @author RT_Borg
+ *
+ * Uses topic id as a proxy for age, which doesn't require loading other pages
+ * to read actual dates.
+ */
+
+export default async function ({ addon, global, console, msg }) {
+  // Default is a little over a month's worth of topics. Reasonably stable through 2022.
+  // There's a lot of tolerance. Half or twice as many topics filter pretty much the same.
+  const TOPICS_PER_MONTH = 7500;
```

Comment:
Would replacing this with a topic count setting (maybe divided by 1000) work so we don't need to maintain this?

Reply:
Back when I started this, I looked at the pace topics increased over the last couple years and found that averaged over months they were surprisingly consistent (IIRC, a small upward drift from one year to the next, but not so much that this constant would need to have changed). I was a surprised there weren't spikes associated with summer or the school year in some way, but I didn't see a dramatic change. It probably deserves a quick yearly spot check, to be on the safe side (or a check if something significantly increases or decreases Scratch use overall).

I'm not sure what alternate age mechanism you're suggesting might be used?

Reply:
I was thinking of replacing the "Roughly how many months is 'old'?" setting with a setting to input the number of topics directly, but if `TOPICS_PER_MONTH` stays relatively stable over time it probably isn't necessary.

Reply:
Funny. That's how I initially implemented it, and earlier reviewers suggested this instead. I think this actually turns out to be much better because if the rate of topics significantly changes, a SA maintainer can make a single change to update it, rather than expecting every user to be knowledgeable enough to adjust their settings.